### PR TITLE
Theming and minor housekeeping

### PR DESCRIPTION
**Pull request by GroovinChip**
This PR contains basic work for theming capabilities in `stream_feed_flutter`. It also contains some minor housekeeping work for the same package.

## CLA

- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required).
- [x] The code changes follow best practices
- [x] Code changes are tested (add some information if not applicable)

### CODE REVIEW

File: packages/stream_feed_flutter/lib/src/utils/extensions.dart [hunk 1]
```diff
@@ -1,43 +1,45 @@
...
 
+  /// TODO: document me
   String str() => <Tag, String>{
         Tag.hashtag: 'hashtag',
         Tag.mention: 'mention',
         Tag.normalText: 'normalText',
       }[this]!;
 
+  /// TODO: document me
   TextStyle style() => <Tag, TextStyle>{
```
Comment (sachaarbonel) [inline packages/stream_feed_flutter/lib/src/utils/extensions.dart:33]: no longer useful I think

File: packages/stream_feed_flutter/lib/src/widgets/comment/item.dart [hunk 2]
```diff
@@ -130,10 +129,11 @@ class CommentItem extends StatelessWidget {
     );
   }
 
+  /// TODO: document me
   List<Widget> displayUsername(User? user) => handleDisplay(
       user?.data,
       nameJsonKey,
-      TextStyle(
+      const TextStyle(
```
Comment (sachaarbonel) [inline packages/stream_feed_flutter/lib/src/widgets/comment/item.dart:136]: Same here add a way to customize username display

File: packages/stream_feed_flutter/lib/src/widgets/dialogs/gif.dart [hunk 3]
```diff
@@ -12,13 +12,15 @@ class GIFAction extends StatelessWidget {
...
-            borderRadius: BorderRadius.circular(6.0),
-            border: Border.all(color: Colors.blue)),
-        child: Icon(
-          Icons.gif_outlined, //TODO: svg icons
-          color: Colors.blue,
-          semanticLabel: 'GIF', //TODO: i18n
-        ));
+      decoration: BoxDecoration(
+        borderRadius: BorderRadius.circular(6),
+        border: Border.all(color: Colors.blue),
```
Comment (sachaarbonel) [inline packages/stream_feed_flutter/lib/src/widgets/dialogs/gif.dart:17]: would be nice to have border-radius and color in the future too. One property for all the dialogs is enough I think

File: packages/stream_feed_flutter/lib/src/widgets/og/card.dart [hunk 4]
```diff
@@ -86,22 +88,22 @@ class ActivityCard extends StatelessWidget {
...
               child: Padding(
-                padding: const EdgeInsets.all(8.0),
+                padding: const EdgeInsets.all(8),
                 child: Column(
                     crossAxisAlignment: CrossAxisAlignment.start,
                     children: [
                       Text(
                         og.title!, //TODO: handle null
-                        style: TextStyle(
+                        style: const TextStyle(
```
Comment (sachaarbonel) [inline packages/stream_feed_flutter/lib/src/widgets/og/card.dart:97]: would be nice to have a way to customize the textStyle of the card too. Something like `cardTitleTextStyle`

File: packages/stream_feed_flutter/lib/src/widgets/og/card.dart [hunk 5]
```diff
@@ -86,22 +88,22 @@ class ActivityCard extends StatelessWidget {
...
                           color: Color(0xff007aff),
                           fontSize: 14,
                           overflow: TextOverflow.ellipsis,
                         ),
                       ),
-                      SizedBox(height: 10),
+                      const SizedBox(height: 10),
                       Text(
                         og.description!, //TODO: handle null
```
Comment (sachaarbonel) [inline packages/stream_feed_flutter/lib/src/widgets/og/card.dart:105]: Same here. Something like `cardDescTextStyle`

File: packages/stream_feed_flutter/lib/src/widgets/user/user_bar.dart [hunk 6]
```diff
@@ -73,62 +77,79 @@ class UserBar extends StatelessWidget {
...
     );
   }
 
+  /// TODO: document me
   List<Widget> displayUsername(User? user) => handleDisplay(
       user?.data,
       nameJsonKey,
-      TextStyle(
-          color: Color(0xff0ba8e0), fontWeight: FontWeight.w700, fontSize: 14));
+      const TextStyle(
```
Comment (sachaarbonel) [inline packages/stream_feed_flutter/lib/src/widgets/user/user_bar.dart:104]: add a way to customize avatar text style in the theme

**Review by sachaarbonel — CHANGES_REQUESTED** [review]
Left a reviews based on our discussion but other than that looks good to me

File: packages/stream_feed_flutter/lib/src/utils/extensions.dart (hunk 1 above)
Comment (GroovinChip) [reply packages/stream_feed_flutter/lib/src/utils/extensions.dart:33]: What in particular is no longer useful?

File: packages/stream_feed_flutter/lib/src/widgets/comment/item.dart (hunk 2 above)
Comment (GroovinChip) [reply packages/stream_feed_flutter/lib/src/widgets/comment/item.dart:136]: I'm not sure of how to do this without adding a whole class just for it, do you have any ideas?

File: packages/stream_feed_flutter/lib/src/widgets/dialogs/gif.dart (hunk 3 above)
Comment (GroovinChip) [reply packages/stream_feed_flutter/lib/src/widgets/dialogs/gif.dart:17]: Can do this with a `GifDialogTheme` and `GifDialogThemeData`

File: packages/stream_feed_flutter/lib/src/utils/extensions.dart (hunk 1 above)
Comment (sachaarbonel) [reply packages/stream_feed_flutter/lib/src/utils/extensions.dart:33]: The extension method style is now handled in InteractiveText via ReactionTheme.of(context).hashtagTextStyle etc. And the method spans is never used I think

File: packages/stream_feed_flutter/lib/src/widgets/dialogs/gif.dart (hunk 3 above)
Comment (sachaarbonel) [reply packages/stream_feed_flutter/lib/src/widgets/dialogs/gif.dart:17]: sounds good

File: packages/stream_feed_flutter/lib/src/utils/extensions.dart (hunk 1 above)
Comment (GroovinChip) [reply packages/stream_feed_flutter/lib/src/utils/extensions.dart:33]: So the whole extension method needs to go?

File: packages/stream_feed_flutter/lib/src/widgets/dialogs/gif.dart (hunk 3 above)
Comment (GroovinChip) [reply packages/stream_feed_flutter/lib/src/widgets/dialogs/gif.dart]: Done

File: packages/stream_feed_flutter/lib/src/widgets/user/user_bar.dart (hunk 6 above)
Comment (GroovinChip) [reply packages/stream_feed_flutter/lib/src/widgets/user/user_bar.dart:104]: Similar problem here as for item.dart

**Review by sachaarbonel — APPROVED** [review]
LGTM

**Review by sachaarbonel — APPROVED** [review]
LGTM

**Pull request merged**